[docs] Removed a pgbouncer limitation (#21173)
yugabyte/yugabyte-db: docs/content/preview/explore/connection-manager/connection-mgr-ysql.md

YugabyteDB inherits the architecture of creating one backend process for every connection to the database from PostgreSQL. These backend processes consume memory and CPU, limiting the number of connections YugabyteDB can support. To solve this problem, you can use a connection pooler, which allows multiplexing multiple client connections to a smaller number of actual server connections, thereby supporting a larger number of connections from applications. [PgBouncer](https://github.com/pgbouncer/pgbouncer) and [Odyssey](https://github.com/yandex/odyssey) are some of the popular PostgreSQL-based server-side connection pooling mechanisms which are fully compatible with YugabyteDB.

> However, these products have a limitation where both Odyssey and PgBouncer do not support SET statements in the transaction pooling mode.

Reviewer: ```suggestion
However, these products have a limitation in that they don't support SET statements in the transaction pooling mode.
```

- Support for session parameters - YSQL Connection Manager supports SET statements, which are not supported by other connection poolers.

> - Support for prepared statements - PgBouncer and Odyssey supports protocol-level prepared statements and YSQL Connection Manager inherits this feature.

Reviewer: ```suggestion
- Support for prepared statements - PgBouncer and Odyssey support protocol-level prepared statements and YSQL Connection Manager inherits this feature.
```
